examples/icmp_response.rs

```diff
@@ -0,0 +1,268 @@
+        .parse::<u32>()
+        .expect("Check chunk count.");
+    let rx_ring_size = (*matches.get_one::<String>("rx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check rx ring size.");
+    let tx_ring_size = (*matches.get_one::<String>("tx_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check tx ring size.");
+    let filling_ring_size = (*matches.get_one::<String>("filling_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check filling ring size.");
+    let completion_ring_size = (*matches.get_one::<String>("completion_ring_size").unwrap())
+        .parse::<u32>()
+        .expect("Check completion ring size.");
+
+    /* signal define to end the application */
+    let term: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
+    let result_sigint: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGINT, Arc::clone(&term));
+    let result_sigterm: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term));
+    let result_sigabrt: Result<SigId, Error> =
+        signal_hook::flag::register(signal_hook::consts::SIGABRT, Arc::clone(&term));
+
+    if result_sigint.or(result_sigterm).or(result_sigabrt).is_err() {
+        panic!("signal is forbidden");
+    }
+
+    /* save source MAC address into variable */
+    let all_interfaces: Vec<NetworkInterface> = interfaces();
+    let is_exist: Option<&NetworkInterface> = all_interfaces
+        .iter()
+        .find(|element| element.name.as_str() == if_name.as_str());
+
+    let src_mac_address: MacAddr = match is_exist {
+        Some(target_interface) => {
+            let option: Option<MacAddr> = target_interface.mac;
+            if option.is_none() {
+                panic!("couldn't find source MAC address");
+            }
+
+            option.unwrap()
+        }
+        None => {
+            panic!("couldn't find source MAC address");
+        }
+    };
+
+    /* execute pv_open() */
+    let pv_open_option: Option<pv::Nic> = pv::pv_open(
+        &if_name,
+        chunk_size,
+        chunk_count,
+        rx_ring_size,
+        tx_ring_size,
+        filling_ring_size,
+        completion_ring_size,
+    );
+    let mut nic: pv::Nic;
+    if let Some(a) = pv_open_option {
+        nic = a;
+    } else {
+        panic!("nic allocation failed!");
+    }
+
+    /* initialize rx_batch_size and packet metadata */
+    let rx_batch_size: u32 = 64;
+    let mut packets: Vec<pv::Packet> = Vec::with_capacity(rx_batch_size as usize);
+
+    while !term.load(Ordering::Relaxed) {
+        let received: u32 = pv::pv_receive(&mut nic, &mut packets, rx_batch_size);
+
+        if received > 0 {
+            let processed: u32 =
+                process_packets(&mut nic, &mut packets, received, &src_mac_address);
+            let sent: u32 = pv::pv_send(&mut nic, &mut packets, processed);
+
+            if sent == 0 {
```

Comment:
Comment that sent always be 0 or equal to processed
Unless that, it seems to be a bug